### PR TITLE
EVG-14656 add patch trigger definitions to API project refs

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-14656](https://jira.mongodb.org/browse/EVG-14656)

### Description 
We neglected to add PatchTriggerAliases to the APIProjectRef, meaning it can't be modified using the API.

### Testing 
Added a unit test.

### TODO
After merging, will add to documentation.

### CODE REVIEW

File: rest/model/project.go [hunk 1]
```diff
@@ -68,7 +54,7 @@ func (t *APITriggerDefinition) BuildFromService(h interface{}) error {
 	switch h.(type) {
 	case model.TriggerDefinition:
 		triggerDef = h.(model.TriggerDefinition)
-	case *model.PeriodicBuildDefinition:
```
Comment (malikchaya2) [inline rest/model/project.go:71]: why do we no longer need case *model.PeriodicBuildDefinition:?

**Review by malikchaya2 — COMMENTED** [review]
@ybrill would it be possible for you to review this as well? I think i remember us discussing this and deciding agains doing it, so I want to make sure I'm not overlooking any potential problems.

**Comment by ybrill** [comment]
@malikchaya2 I don't remember a reason not to do this. I probably just overlooked it when I was adding it to the ProjectRef.

File: rest/model/project.go (hunk 1 above)
Comment (ablack12) [reply rest/model/project.go:71]: This was a copypaste error. We wouldn't be converting a periodic build type to a trigger type.

**Review by malikchaya2 — APPROVED** [review]

**Pull request merged**